src/main/java/org/jboss/qa/phaser/processors/MethodExecutor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015 Red Hat Inc. and/or its affiliates and other contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.qa.phaser.processors;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import lombok.Builder;
+import lombok.Singular;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Builder
+public class MethodExecutor {
+
+	private ParameterProcessor defaultProcessor;
+
+	@Singular
+	private List<ParameterProcessor> processors;
+
+	public void invokeMethod(Method method, Object target) throws InvocationTargetException, IllegalAccessException {
+		try {
+			final Class<?>[] paramClasses = method.getParameterTypes();
+			if (paramClasses.length == 0) { // no parameters to inject
+				if (log.isDebugEnabled()) {
+					log.debug("Execute method {}#{}", target.getClass().getCanonicalName(), method.getName());
+				}
+				method.invoke(target);
+				return;
+			}
+
+			final Annotation[][] paramAnnotations = method.getParameterAnnotations();
+			final ArrayList<List<Object>> params = new ArrayList<>(paramClasses.length);
+			params.ensureCapacity(paramClasses.length);
+
+			for (int i = 0; i < paramClasses.length; i++) {
+				for (int j = 0; j < paramAnnotations[i].length; j++) {
+					final Annotation annotation = paramAnnotations[i][j];
+					for (ParameterProcessor processor : processors) {
+						if (processor.getAnnotationClass().isAssignableFrom(annotation.getClass())) {
+							params.add(i, processor.processParameter(paramClasses[i], annotation));
```

Comment:
@osmman If you want to use more processors for one parameter it will add new parameter for each processor and then you will get exception:
```
Caused by: java.lang.IllegalArgumentException: wrong number of arguments
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.jboss.qa.phaser.processors.MethodExecutor.invokeMethod(MethodExecutor.java:77)
	... 28 more
```
```java
@PhaseA
public void method(@Property("prop1") @Inject CharSequence url) { }
```